main/test/ca/uwaterloo/flix/language/phase/TestLexer.scala

```diff
@@ -0,0 +1,277 @@
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedBlockComment](result)
+  }
+
+  test("LexerError.UnterminatedBlockComment.03") {
+    val input =
+      s"""
+         |def f(): Unit = ()
+         |/* This is unterminated
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedBlockComment](result)
+  }
+
+  test("LexerError.UnterminatedBlockComment.04") {
+    val input =
+      s"""
+         | /* /* This is unterminated and nested */
+         |def f(): Unit = ()
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedBlockComment](result)
+  }
+
+  test("LexerError.UnterminatedBuiltIn.01") {
+    val input =
+      s"""
+         |def f(): Int32 = $$BUILT_IN; 42
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedBuiltIn](result)
+  }
+
+  test("LexerError.UnterminatedBuiltIn.02") {
+    val input =
+      s"""
+         |def f(): Unit = $$BUILT_IN
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedBuiltIn](result)
+  }
+
+  test("LexerError.UnterminatedBuiltIn.03") {
+    val input =
+      s"""
+         |def f(): Unit = $$BUILT_/*IN*/$$
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedBuiltIn](result)
+  }
+
+  test("LexerError.UnterminatedBuiltIn.04") {
+    val input =
+      s"""
+         |def f(): Unit = $$INT64_REM$$
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[ParseError](result)
+  }
+
+  test("LexerError.UnterminatedChar.01") {
+    val input =
+      s"""
+         |def f(): Char = 'a
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedChar](result)
+  }
+
+  test("LexerError.UnterminatedChar.02") {
+    val input =
+      s"""
+         |def f(): Char = '\uffff
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedChar](result)
+  }
+
+  test("LexerError.UnterminatedChar.03") {
+    val input =
+      s"""
+         |def f(): Char = 'a/* This is a block-comment */'
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedChar](result)
+  }
+
+  test("LexerError.UnterminatedChar.04") {
+    val input =
+      s"""
+         |def f(): Char = '/* This is a block-comment */a'
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedChar](result)
+  }
+
+  test("LexerError.UnterminatedInfixFunction.01") {
+    val input =
+      s"""
+         |def f(): Int32 = 1 `add 2
+       """.stripMargin
```

Comment:
Notice that you have a new line in all these tests.

Maybe you want to double the tests s.t. you also check for un-terminated XYZ when the file ends on the same line.